feat(curriculum): add Working with npm Scripts theory block (#64221)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/blocks/lecture-working-with-npm-scripts/692aba0978ca391f10455b4a.md

```bash
npm init --scope=@<scope-name>
```

The scope name can be your username if you are using a personal account, or the name of your organization if you are using an organizational account.

## Step 4: Create README.md

The README of your package is a markdown file where you describe what your package does and its intended usage.

Some examples of important content that you may want to include are:

* Its purpose.
    
* How to install it.
    
* Important dependencies and considerations.
    
* How to get started.
    
* Sample code snippets.
    
* Contribution guidelines (if you accept contributions).
    
* Link to a changelog file where you track the changes that have been made in different versions.
    

The structure of the README is flexible, though. You can adapt it to fit your needs.

## **Step 5:** `.npmignore`

The `.npmignore` file tells the npm registry what files to include and what files to ignore when publishing your package. It's very important to prevent sharing sensitive information such as private keys, passwords, and personal information.

> In this file, you can write patterns to specify what files, directories, and types of files are ignored. It follows the same pattern matching rules as the `.gitignore` files.

Reviewer: ```suggestion
In this file, you can write patterns to specify what files, directories, and types of files are ignored. It follows the same pattern-matching rules as the `.gitignore` files.
```

The README of your package is a markdown file where you describe what your package does and its intended usage.

Some examples of important content that you may want to include are:

* Its purpose.
    
* How to install it.
    
* Important dependencies and considerations.
    
* How to get started.
    
* Sample code snippets.
    
* Contribution guidelines (if you accept contributions).
    
* Link to a changelog file where you track the changes that have been made in different versions.
    

The structure of the README is flexible, though. You can adapt it to fit your needs.

## **Step 5:** `.npmignore`

The `.npmignore` file tells the npm registry what files to include and what files to ignore when publishing your package. It's very important to prevent sharing sensitive information such as private keys, passwords, and personal information.

In this file, you can write patterns to specify what files, directories, and types of files are ignored. It follows the same pattern matching rules as the `.gitignore` files. You can learn more about these rules in the [official Git documentation](https://git-scm.com/book/en/v2/Git-Basics-Recording-Changes-to-the-Repository#_ignoring).

> If you have worked with Git, you may think that this file sounds very similar to `.gitignore`, right? Well, that's because it is. According to the [npm documentation](https://docs.npmjs.com/cli/v9/using-npm/developers), "If there's no `.npmignore` file, but there is a `.gitignore` file, then npm will ignore the stuff matched by the `.gitignore` file."

Reviewer 2: ```suggestion
In this file, you can write patterns to specify what files, directories, and types of files are ignored. It follows the same pattern matching rules as the `.gitignore` files.

If you have worked with Git, you may think that this file sounds very similar to `.gitignore`, right? Well, that's because it is. According to the npm documentation, "If there's no `.npmignore` file, but there is a `.gitignore` file, then npm will ignore the stuff matched by the `.gitignore` file."
```

Generally, we don't like links to external sites. I would vote to just remove them, but I would be fine leaving them if we think they're necessary. I think it works without them.

```bash
npm publish --access public
```

To see your public package, go to this URL:

```bash
https://npmjs.com/package/<your-package-name>
```

You should see an indication of the visibility of your package (public or private) below the package name.

## **Step 7 (Optional): Install Your Package**

Now that you are all set up, you can install your package in a Node.js project with this command:

```bash
npm install <your-package-name>
```

## Updating Your Package

If you need to make changes to your package after publishing it, you can update the package version in `package.json`.

This is the [standard convention](https://docs.npmjs.com/about-semantic-versioning#incrementing-semantic-versions-in-published-packages) for semantic versioning:

* **First release:** 1.0.0.
    
* **Patch release:** Increment the third digit. For example, updating from 1.0.0 to 1.0.1.
    
* **Minor release:** Increment the middle digit and reset the last digit to zero. For example, updating 1.0.0 to 1.1.0.
    

> * **Major release:** Increment the first digit and reset the middle and last digits to zero. For example, updating 1.1.0 to 2.0.0.

Reviewer 2: Maybe put all the versions here in backticks and quotes? Probably fine as-is if we want.

    

Great work. Now you know how to publish your package to the npm registry and how to update it to fix bugs and add new features.

# --questions--

## --text--

What is the command used in the terminal to upload a package from your local machine to the npm registry?

## --answers--

`npm install <package-name>`

### --feedback--

This is the command that finalizes the process, making your package available for others to `npm install`.

---

`npm start`

### --feedback--

This is the command that finalizes the process, making your package available for others to `npm install`.

---

`npm add`

### --feedback--

> This is the command that finalizes the process, making your package available for others to `npm install`.

Reviewer 2: ```suggestion
This is the command that finalizes the process, making your package available for others to install.

---

`npm start`

### --feedback--

This is the command that finalizes the process, making your package available for others to install.

---

`npm add`

### --feedback--

This is the command that finalizes the process, making your package available for others to install.
```

Pretty minor, but this reads a little better to me.
